packages/calypso-e2e/src/lib/utils/get-test-account.ts

```diff
@@ -0,0 +1,70 @@
+type Env = 'edge' | 'production';
+
+type SiteType = 'simple' | 'atomic';
+
+type DeepPartial< T > = T extends object
+	? {
+			[ P in keyof T ]?: DeepPartial< T[ P ] >;
+	  }
+	: T;
+
+type Plugin = 'gutenberg' | 'coblocks';
+
+type Accounts = { [ key in SiteType ]: { [ key in Env ]: string } };
+type AccountsIndex = { [ key in Plugin ]: Accounts };
+
+type OverridenAccounts = DeepPartial< Accounts >;
+
+type Options = {
+	isEdge?: boolean;
+	isAtomic?: boolean;
+	overridenAccounts?: OverridenAccounts;
+};
+
+const defaultGutenbergAccounts: Accounts = {
+	simple: { edge: 'gutenbergSimpleSiteEdgeUser', production: 'gutenbergSimpleSiteUser' },
+	atomic: { edge: 'gutenbergAtomicSiteEdgeUser', production: 'eCommerceUser' },
+};
+
+const defaultCoblocksAccounts: Accounts = {
+	simple: { edge: 'coBlocksSimpleSiteEdgeUser', production: 'gutenbergSimpleSiteUser' },
+	// not being used, we don't yet test coblocks edge on AT atm
+	atomic: { edge: 'gutenbergAtomicSiteEdgeUser', production: 'eCommerceUser' },
+};
+
+const accountsIndex: AccountsIndex = {
+	gutenberg: defaultGutenbergAccounts,
+	coblocks: defaultCoblocksAccounts,
+};
+
+function getTestSite(
+	forPlugin: Plugin,
+	options: Options = {
+		isEdge: false,
+		isAtomic: false,
+	}
+) {
+	const { isEdge, isAtomic, overridenAccounts } = options;
+
+	let accounts = accountsIndex[ forPlugin ];
+
+	const env = isEdge ? 'edge' : 'production';
+	const target = isAtomic ? 'atomic' : 'simple';
+
+	if ( overridenAccounts ) {
+		for ( let key in accounts ) {
+			const target: SiteType = key as SiteType;
+			accounts[ target ] = { ...accounts[ target ], ...overridenAccounts[ target ] };
+		}
+	}
+
+	return accounts[ target ][ env ];
+}
+
+export function getTestAccountForGutenberg( options: Options ) {
+	return getTestSite( 'gutenberg', options );
+}
+
+export function getTestAccountForCoblocks( options: Options ) {
+	return getTestSite( 'coblocks', options );
+}
```

Comment:
I think having dedicated methods that fill in part of the required details (like these) are fine as long as the combinations don't grow too much.

I tend to prefer having a generic method/function that can handle different scenarios, so something like:

```
export function getTestAccountByFeature( plugin: string, options: Options ) {
}

```

At that point it might make sense to rename the `getTestSite` function into `getTestAccountByFeature` so callers don't have to pass though two layers.

This is a stylistic suggestion; the current implementation is perfectly suitable for now, I think.

Reply:
I like the idea of making it more general and by feature. In the future, we might use this feature to select specific sites for other plugins or even other components. I'll play with it a bit.